### PR TITLE
AM-559-Test Case for Upload document(binary content)

### PR DESCRIPTION
**Pull request by shrivsksavit**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/AM-559

### Change description ###

Test case - 'Upload document(binary content) to Doc-store' API at case-document micro-service

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/functionalTest/resources/features/F-004/F-000.feature [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  @S-041
+  Scenario: must successfully Post Upload Document with Binary Content
+    Given a user with [an active caseworker profile in CCD with full permissions on a document field]
+    When  a request is prepared with appropriate values
+    And   it is submitted to call the [Post Upload Document with Binary Content] operation of [CCD Case Document AM API]
+    Then  a positive response is received
+    And   the response [contains the metadata for the document uploaded above]
+    And   the response has all other details as expected
+
+  @S-042
```
Comment (aashishaggarwal123) [inline src/functionalTest/resources/features/F-004/F-000.feature]: With in the scope of this upload API that covers only step number 2,3,4,5 &6 as per the LLD, validating the existence of casteType, JID, user-roles and user-id is not possible and hence these scenarios are out of scope.

File: src/functionalTest/resources/features/F-004/F-000.feature [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  Scenario: must get an error response for a malformed classification
+    Given a user with [an active caseworker profile in CCD with full permissions on a document field]
+    When  a request is prepared with appropriate values
+    And   the request [for a malformed classification]
+    And   it is submitted to call the [Post Upload Document with Binary Content] operation of [CCD Case Document AM API]
+    Then  a negative response is received
+    And   the response has all the details as expected
+
+  @S-048
+  Scenario: must get an error response for a non-existing roles
```
Comment (aashishaggarwal123) [inline src/functionalTest/resources/features/F-004/F-000.feature]: same as above.

File: src/functionalTest/resources/features/F-004/F-000.feature [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  @S-051
+  Scenario: must get an error response for a malformed roles
+    Given a user with [an active caseworker profile in CCD with full permissions on a document field]
+    When  a request is prepared with appropriate values
+    And   the request [for a malformed roles]
+    And   it is submitted to call the [Post Upload Document with Binary Content] operation of [CCD Case Document AM API]
+    Then  a negative response is received
+    And   the response has all the details as expected
+
+  @S-052
```
Comment (aashishaggarwal123) [inline src/functionalTest/resources/features/F-004/F-000.feature]: We can't check the user-id even if the user is not having create permission on the case, he would be able to upload document with 10 minute TTL that will delete the document because ccd side create/update event will fail due to insufficient permission.

File: src/functionalTest/resources/features/F-004/F-000.feature (hunk 1 above)
Comment (shrivsksavit) [reply src/functionalTest/resources/features/F-004/F-000.feature]: removed this test

File: src/functionalTest/resources/features/F-004/F-000.feature (hunk 2 above)
Comment (shrivsksavit) [reply src/functionalTest/resources/features/F-004/F-000.feature]: removed

File: src/functionalTest/resources/features/F-004/F-000.feature (hunk 3 above)
Comment (shrivsksavit) [reply src/functionalTest/resources/features/F-004/F-000.feature]: Ignore this test and put the comment that this test will be done manually

File: src/functionalTest/resources/features/F-001/F-001.feature [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  @S-003
+  Scenario: must get an error response for a malformed document ID
+    Given a user with [an active caseworker profile in CCD with full permissions on a document field]
+    When  a request is prepared with appropriate values
+    And   the request [for a malformed document ID]
+    And   it is submitted to call the [Get Document Metadata by Document ID] operation of [CCD Case Document AM API]
+    Then  a negative response is received
+    And   the response has all the details as expected
+
+  @S-004
```
Comment (aashishaggarwal123) [inline src/functionalTest/resources/features/F-001/F-001.feature]: S-004 won't not be possible in this APIs functional testing scope as it is dependent upon the following element after integration with AM-450:
1. The document need to uploaded first.
2. The document metadata must be linked with an existing ccd case reference.
3. CCD case must be having the same document Id (in documentFiled>url) which is uploaded in the step #1 above.

Overall it this expects that a document must be uploaded as per the updated upload process which is very difficult to simulate for testing this API, therefore this scenario can only be tested manually. we can evaluate the situation again after full upload(3 legs) implementation.

File: src/functionalTest/resources/features/F-004/F-004.feature [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  @S-046
+  Scenario: must get an error response for a malformed classification
+    Given a user with [an active caseworker profile in CCD with full permissions on a document field]
+    When  a request is prepared with appropriate values
+    And   the request [for a malformed classification]
+    And   it is submitted to call the [Post Upload Document with Binary Content] operation of [CCD Case Document AM API]
+    Then  a negative response is received
+    And   the response has all the details as expected
+
+  @S-047
```
Comment (aashishaggarwal123) [inline src/functionalTest/resources/features/F-004/F-004.feature]: roles parameter might be removed from our API spec. so this scenario will become obsolete later.

File: src/functionalTest/resources/features/F-004/F-004.feature [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    When  a request is prepared with appropriate values
+    And   the request [contain multiple document with binary content]
+    And   it is submitted to call the [Post Upload Document with Binary Content] operation of [CCD Case Document AM API]
+    Then  a positive response is received
+    And   the response [contains the metadata for the document uploaded above]
+    And   the response has all other details as expected
+
+  @S-050
+  Scenario: must get an error response for a above max allowed size of a document
+    Given a user with [an active caseworker profile in CCD with full permissions on a document field]
```
Comment (aashishaggarwal123) [inline src/functionalTest/resources/features/F-004/F-004.feature]: I think, these scenarios are not updated on JIRA. please do that. considering JIRA 578 already merged, we should be able to run all and finalise the test data for all the scenarios of upload API.

File: src/functionalTest/resources/features/F-004/F-004.feature (hunk 5 above)
Comment (shrivsksavit) [reply src/functionalTest/resources/features/F-004/F-004.feature]: at the moment we will keep this once its conformed will removed

File: src/functionalTest/resources/features/F-004/F-004.feature (hunk 6 above)
Comment (shrivsksavit) [reply src/functionalTest/resources/features/F-004/F-004.feature]: added in jira
